### DISCUSSION TITLE
Can duckdb directly load data from JSON objects?

### DISCUSSION THREAD

**Discussion by hw2499** [duckdb/duckdb#7051] [General]
1、Can duckdb directly load data from JSON objects,

For example:

json_ object = '[{"name":"abc","age":"20"},{"name":"def","age":"32"}]'

select name,age from json_ object

2、 What OLAP functions does duck db support?

**Comment by tadpol** [comment]
From the docs: https://duckdb.org/docs/extensions/json

**Reply by hw2499** [reply]
Not reading from a file, but a JSON object in memory

**Reply by tadpol** [reply]
🤔; https://duckdb.org/docs/extensions/json#json-scalar-functions 

Maybe this?
```sql
select json('[{"name":"abc","age":"20"},{"name":"def","age":"32"}]')->'$[0].name';
```

I don't see a `json_to_recordset` function. But it looks like you can do something like:
```sql
select a->'name',a->'age' from unnest(CAST(json('[{"name":"abc","age":"20"},{"name":"def","age"::"32"}]') as varchar[])) as a(a);
```

**Reply by Tishj** [reply]
> 1、Can duckdb directly load data from JSON objects,
> 
> For example:
> 
> json_ object = '[{"name":"abc","age":"20"},{"name":"def","age":"32"}]'
> 
> select name,age from json_ object
> 
> 2、 What OLAP functions does duck db support?

Are you asking about the Python client?
Support is planned in the `DuckDBPyConnection.read_json` method for reading from file-like objects.
The `read_csv` method already supports this

**Reply by hw2499** [reply]
golang client
 Yes, the result set in memory is jsonobject. I hope you can directly perform select queries from this result set 

> 🤔; https://duckdb.org/docs/extensions/json#json-scalar-functions
> 
> Maybe this?
> 
> ```sql
> select json('[{"name":"abc","age":"20"},{"name":"def","age":"32"}]')->'$[0].name';
> ```
> 
> I don't see a `json_to_recordset` function. But it looks like you can do something like:
> 
> ```sql
> select a->'name',a->'age' from unnest(CAST(json('[{"name":"abc","age":"20"},{"name":"def","age"::"32"}]') as varchar[])) as a(a);
> ```

 Yes, the result set in memory is jsonobject. I hope you can directly perform select queries from this result set

**Reply by hw2499** [reply]
> > 1、Can duckdb directly load data from JSON objects,
> > For example:
> > json_ object = '[{"name":"abc","age":"20"},{"name":"def","age":"32"}]'
> > select name,age from json_ object
> > 2、 What OLAP functions does duck db support?
> 
> Are you asking about the Python client? Support is planned in the `DuckDBPyConnection.read_json` method for reading from file-like objects. The `read_csv` method already supports this

golang client，Json Object is in memory and not stored locally.
